src/test/java/nextstep/step4/domain/PointTest.java

```diff
@@ -0,0 +1,51 @@
+package nextstep.step4.domain;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+class PointTest {
+
+    @Test
+    @DisplayName("처음 위치하는 점에서 move 메서드 호출 시 이동한 위치")
+    void firstPointMove() {
+        assertThat(Point.first(false).move()).isEqualTo(0);
+        assertThat(Point.first(true).move()).isEqualTo(1);
+    }
+
+    @Test
+    @DisplayName("2번째에 오는 점에서 양방향으로 이동할 수 없을 때 가만히 있어야 함")
+    void nextPointStay() {
+        Point secondPoint = Point.first(false).next(false);
+        assertThat(secondPoint.move()).isEqualTo(1);
+    }
+
+    @Test
+    @DisplayName("2번째에 오는 점이 isLeft 가능할 경우 움직였을 때 처음 column 으로 이동해야 함")
+    void nextPointLeftMove() {
+        Point secondPoint = Point.first(true).next(false);
+        assertThat(secondPoint.move()).isEqualTo(0);
+    }
+
+    @Test
+    @DisplayName("2번째에 오는 점이 isRight 가능할 경우 움직였을 때 세 번째 column 으로 이동해야 함")
+    void nextPointRightMove() {
+        Point secondPoint = Point.first(false).next(true);
+        assertThat(secondPoint.move()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("처음 위치 isRight, 두 번째 위치를 Generator 를 사용하여도 좌측으로 이동하게 만들어짐")
+    void randomNextPointMoveLeft() {
+        Point secondPoint = Point.first(true).next();
+        assertThat(secondPoint.move()).isEqualTo(0);
```

Comment:
```suggestion
        assertThat(secondPoint.move()).isZero();
```
IDE에서 제안하길 이런형태도 가능하다고 하네요 😄